### DISCUSSION TITLE
Enable multiple proxies in settings.yml

### DISCUSSION THREAD

**Discussion by paradoxsupreme** [searx/searx#2576] [Installation issue: custom]
I tried to enable multiple proxies so I added these in my settings.yml:
```outgoing: # communication with search engines
  proxies:
    http:
      - http://proxy1:8080
      - http://proxy2:8080
    https:
      - http://proxy1:8080
      - http://proxy2:8080
```
but I get internal server error. Do I have to do something else ?
Do I have to use some proxy server in background like in [here](https://github.com/searx/searx/pull/2319)?

**Comment by unixfox** [comment]
The point of using a proxy it's to do requests from another public IP address. Please consult wikipedia on what is a proxy: https://en.wikipedia.org/wiki/Proxy_server

If you run the proxy on the same exact public IP address then it won't fix your issue with Google captcha because Google will see no difference between Searx and your proxy due to the fact that it's under the same public IP address.

You can find free public proxies on the internet, but usually they are already blocked by Google. That's why it's better to use a paid service, but I won't disclose any here because I don't want to make any advertising.

**Reply by paradoxsupreme** [reply]
I think I understand now. Thanks!

**Reply by paradoxsupreme** [reply]
> The point of using a proxy it's to do requests from another public IP address. Please consult wikipedia on what is a proxy: https://en.wikipedia.org/wiki/Proxy_server
> 
> If you run the proxy on the same exact public IP address then it won't fix your issue with Google captcha because Google will see no difference between Searx and your proxy due to the fact that it's under the same public IP address.
> 
> You can find free public proxies on the internet, but usually they are already blocked by Google. That's why it's better to use a paid service, but I won't disclose any here because I don't want to make any advertising.

Regarding paid proxies I have some more questions:
- Can I use a vps or two to self host a proxy server or do I have to use rotating proxies ?
- How many proxies do I have to use in order for google to work properly ?

**Reply by unixfox** [reply]
> > The point of using a proxy it's to do requests from another public IP address. Please consult wikipedia on what is a proxy: [en.wikipedia.org/wiki/Proxy_server](https://en.wikipedia.org/wiki/Proxy_server)
> > If you run the proxy on the same exact public IP address then it won't fix your issue with Google captcha because Google will see no difference between Searx and your proxy due to the fact that it's under the same public IP address.
> > You can find free public proxies on the internet, but usually they are already blocked by Google. That's why it's better to use a paid service, but I won't disclose any here because I don't want to make any advertising.
> 
> Regarding paid proxies I have some more questions:
> 
>     * Can I use a vps or two to self host a proxy server or do I have to use rotating proxies ?
> 
>     * How many proxies do I have to use in order for google to work properly ?

Yes you can use a VPS for that.

And I don't know the exact amount of proxies to use for bypassing Google captcha. Just already start with one, and you will see if you need more.

**Reply by paradoxsupreme** [reply]
> > > The point of using a proxy it's to do requests from another public IP address. Please consult wikipedia on what is a proxy: [en.wikipedia.org/wiki/Proxy_server](https://en.wikipedia.org/wiki/Proxy_server)
> > > If you run the proxy on the same exact public IP address then it won't fix your issue with Google captcha because Google will see no difference between Searx and your proxy due to the fact that it's under the same public IP address.
> > > You can find free public proxies on the internet, but usually they are already blocked by Google. That's why it's better to use a paid service, but I won't disclose any here because I don't want to make any advertising.
> > 
> > 
> > Regarding paid proxies I have some more questions:
> > ```
> > * Can I use a vps or two to self host a proxy server or do I have to use rotating proxies ?
> > 
> > * How many proxies do I have to use in order for google to work properly ?
> > ```
> 
> Yes you can use a VPS for that.
> 
> And I don't know the exact amount of proxies to use for bypassing Google captcha. Just already start with one, and you will see if you need more.


I used proxy.py on the same VPS that I am hosting searx and I added the proxy address in settings.yml and google is working now. But how is it possible ? Isn's the same as running the searx instance without proxy ? Because it is the same IP.

**Reply by unixfox** [reply]
Try without it. Maybe you could unblocked while you were trying to setup your proxy.

**Reply by paradoxsupreme** [reply]
> Try without it. Maybe you could unblocked while you were trying to setup your proxy.

That's it. It is also working without.
